### PR TITLE
Use signatures in WebAPI::Controller::Step

### PR DESCRIPTION
**Pull request by okurz**
* Use signatures in WebAPI::Controller::Step
* Use signatures in WebAPI::Controller::Running
* Use signatures in WebAPI::Controller::Main
* Use signatures in WebAPI::Controller::File
* Use signatures in WebAPI::Controller::Developer
* Use signatures in WebAPI::Controller::ApiKey

After:
* https://github.com/os-autoinst/openQA/pull/4882

### CODE REVIEW

File: lib/OpenQA/WebAPI/Controller/Running.pm [hunk 1]
```diff
@@ -79,13 +75,11 @@ sub edit {
...
-sub streamtext {
-    my ($self, $file_name, $start_hook, $close_hook) = @_;
-
+sub streamtext ($self, $file_name, $start_hook, $close_hook) {
     my $job = $self->stash('job');
     my $worker = $job->worker;
-    $start_hook ||= sub { };
-    $close_hook ||= sub { };
+    $start_hook ||= sub () { };
+    $close_hook ||= sub () { };
```
Comment (perlpunk) [inline lib/OpenQA/WebAPI/Controller/Running.pm:82]: The hooks are called with parameters, so this should fail.

File: lib/OpenQA/WebAPI/Controller/Running.pm [hunk 2]
```diff
@@ -79,13 +75,11 @@ sub edit {
     $self->redirect_to('edit_step', moduleid => $running_module->name(), stepid => $stepid);
 }
 
-sub streamtext {
-    my ($self, $file_name, $start_hook, $close_hook) = @_;
-
+sub streamtext ($self, $file_name, $start_hook, $close_hook) {
```
Comment (perlpunk) [inline lib/OpenQA/WebAPI/Controller/Running.pm:78]: `Too few arguments for subroutine 'OpenQA::WebAPI::Controller::Running::streamtext' at t/26-controllerrunning.t line 76.`

File: lib/OpenQA/WebAPI/Controller/Step.pm [hunk 3]
```diff
@@ -250,9 +241,7 @@ sub edit {
     $self->render('step/edit');
 }
 
-sub _new_screenshot {
-    my ($self, $tags, $image_name, $matches) = @_;
-
+sub _new_screenshot ($self, $tags, $image_name, $matches) {
```
Comment (perlpunk) [inline lib/OpenQA/WebAPI/Controller/Step.pm:244]: `$matches` is optional

File: lib/OpenQA/WebAPI/Controller/Running.pm [hunk 4]
```diff
@@ -118,14 +112,14 @@ sub streamtext {
     # Check for new lines from the logfile using recurring timer
     # Setup utility function to close the connection if something goes wrong
     my $timer_id;
-    my $close_connection = sub {
+    my $close_connection = sub ($self) {
```
Comment (okurz) [inline lib/OpenQA/WebAPI/Controller/Running.pm:115]: What do you guys think: Is this function just not properly collected in coverage or is it unused code?

**Review by Martchus — APPROVED** [review]

File: lib/OpenQA/WebAPI/Controller/Running.pm (hunk 4 above)
Comment (okurz) [reply lib/OpenQA/WebAPI/Controller/Running.pm]: excluding "Running" from this PR for now.

Comment (okurz) [reply lib/OpenQA/WebAPI/Controller/Running.pm]: -> https://github.com/os-autoinst/openQA/pull/4898

**Review by Martchus — APPROVED** [review]

**Review by perlpunk — APPROVED** [review]

**Pull request merged**